### PR TITLE
Clouds.yaml, no_gateway_ip, subnetpool allocation

### PR DESCRIPTION
**Pull request by markgoddard**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -41,6 +43,8 @@ dict containing the following items:
    - `cidr`: CIDR representation of the neutron subnet's IP network.
    - `dns_nameservers`: A list of DNS nameservers for the subnet.
    - `gateway_ip`: IP address of the neutron subnet's gateway.
+   - `no_gateway_ip`: Optional boolean, whether to omit a gateway IP. If unset,
+     this will be `true` if `gateway_ip` is specified, and `false` otherwise.
```
Comment (oneswig) [inline README.md:47]: Isn't it sufficient just to set a value for gateway IP or not?  What does it mean if `gateway_ip` is defined but `no_gateway_ip` is true?  Or is this to make up a sensible default if neither parameter is set?

**Review by oneswig — COMMENTED** [review]
Just a question on the truth table of `gateway_ip` vs `no_gateway_ip`

File: README.md (hunk 1 above)
Comment (markgoddard) [reply README.md:47]: The case I want to allow is where we want a gateway, but don't want to specify what its IP will be. This is necessary for subnet pools. I'll let the module handle the case where gateway_ip is defined and no_gateway_ip is true.

**Comment by markgoddard** [comment]
Added another commit that supports deleting networks with routers on them.

**Review by oneswig — APPROVED** [review]
lgtm

**Pull request merged**